packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1304,6 +1348,33 @@ export class MergeTree {
 					segment: pendingSegment,
 				});
 			});
+
+			if (opArgs.op.type === MergeTreeDeltaType.REMOVE) {
+				// if segments slid to the right, then we have to slide pending
+				// segments in reverse order
+				const newSegment = this._getSlideToSegment(pendingSegmentGroup.segments[0]);
+				const shouldReverse =
+					pendingSegmentGroup.segments[0] &&
+					newSegment &&
+					pendingSegmentGroup.segments[0].ordinal < newSegment.ordinal;
+
+				const pendingSegments = shouldReverse
```

Comment:
there's a lot of unnecessary array copying in this implementation. it seems simpler to just use a for loop with an explicit index.

Reply:
also, the fact that we need to reverse to preserve ref order for undo-redo makes me think we actually have a lurking bug in the other call to `slideAckedRemovedSegmentReference` in the overlapping remove case. Repro would be something like create "12" with each in its own segment, remove it locally, mark "12" removed remotely before the local ack comes back, then undo locally.

this line in particular:

Reply:
if you agree with ^, i'll file a bug to investigate. definitely not caused by current change

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/4031